packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -601,6 +651,47 @@ export class Cluster extends Resource implements ICluster {
     tagAllSubnets('private', this.vpc.privateSubnets, "kubernetes.io/role/internal-elb");
     tagAllSubnets('public', this.vpc.publicSubnets, "kubernetes.io/role/elb");
   }
+
+  /**
+   * Patches the CoreDNS deployment configuration and sets the "eks.amazonaws.com/compute-type"
+   * annotation to either "ec2" or "fargate". Note that if "ec2" is selected, the resource is
+   * omitted/removed, since the cluster is created with the "ec2" compute type by default.
+   */
+  private defineCoreDnsComputeType(type: CoreDnsComputeType) {
+    if (!this.kubectlEnabled) {
```

Comment:
Is this worth an error?

Reply:
it's a private API, but ok, I've wrapped the outer call with a condition and added an error here.